src/seedu/addressbook/commands/Command.java

```diff
@@ -54,6 +54,12 @@ public void setData(AddressBook addressBook, List<? extends ReadOnlyPerson> rele
         this.relevantPersons = relevantPersons;
     }
 
+    public boolean isMutating(){
+        return (this.getClass().getName().equals("AddCommand") ||
+                this.getClass().getName().equals("DeleteCommand") ||
+                this.getClass().getName().equals("ClearCommand"));
+    }
+
```

Comment:
There is a way for you not to hard-code this method in the parent class and make every child node must implement itself.